### PR TITLE
feat: add csai demo

### PR DESCRIPTION
**Pull request by hongfeih-es**
add csai demo and doc

- [DORIS-2117](https://dicetech.atlassian.net/browse/DORIS-2117)

[DORIS-2117]: https://dicetech.atlassian.net/browse/DORIS-2117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,6 +2,7 @@
 plugins {
     id 'com.android.application' version '7.2.0' apply false
     id 'com.android.library' version '7.2.0' apply false
+    id 'org.jetbrains.kotlin.android' version '1.9.22' apply false
```
Comment (guoen21) [inline build.gradle:5]: our doris-android use 1.8 and mobile app use 1.7, not sure we should use 1.9 or lower on example app.

File: video/build.gradle [hunk 2]
```diff
@@ -31,9 +31,15 @@ android {
...
 
     implementation "androidx.appcompat:appcompat:$androidxAppCompatVersion"
-    implementation "com.google.android.material:material:$androidxMaterialVersion"
     implementation "androidx.constraintlayout:constraintlayout:$androidxConstraintLayoutVersion"
+    implementation "com.google.android.material:material:$androidxMaterialVersion"
 
-    implementation "com.github.DiceTechnology.doris-android:doris:$dorisVersion"
-    implementation "com.github.DiceTechnology.doris-android:doris-ui:$dorisVersion"
+    implementation("com.github.DiceTechnology.doris-android:doris:$dorisVersion") {
+        exclude group: 'com.github.DiceTechnology', module: 'dice-shield-android'
```
Comment (guoen21) [inline video/build.gradle]: we can remove this line

File: video/build.gradle [hunk 3]
```diff
@@ -31,9 +31,15 @@ android {
...
+        exclude group: 'androidx.media3'
+    }
+    implementation("com.github.DiceTechnology.doris-android:doris-ui:$dorisVersion") {
+        exclude group: 'com.github.DiceTechnology', module: 'dice-shield-android'
```
Comment (guoen21) [inline video/build.gradle]: we can remove this line

**Review by guoen21 — DISMISSED** [review]

**Review by szaboa — DISMISSED** [review]

**Review by guoen21 — DISMISSED** [review]

File: csai/src/main/java/com/dice/doris/csai/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+    private fun createPlayer(adType: DorisAdEvent.AdType): ExoDoris {
+        val builder = if (adType === DorisAdEvent.AdType.IMA_CSAI) {
+            ExoDorisImaCsaiBuilder(this@MainActivity).apply {
+                setAdViewProvider(playerView)
+            }
+        } else if (adType === DorisAdEvent.AdType.IMA_CSAI_LIVE) {
+            ExoDorisImaCsaiLiveBuilder(this@MainActivity).apply {
+                setLiveAdSurfaceView(playerView.videoSurfaceView as SurfaceView)
+                setAdViewProvider(playerView)
```
Comment (guoen21) [inline csai/src/main/java/com/dice/doris/csai/MainActivity.kt]: For ima csai live player, we need two player views, one for content playback, second for ad playback.

**Review by guoen21 — CHANGES_REQUESTED** [review]

File: csai/src/main/java/com/dice/doris/csai/MainActivity.kt (hunk 4 above)
Comment (guoen21) [reply csai/src/main/java/com/dice/doris/csai/MainActivity.kt]: try to fix it with #13

**Review by guoen21 — DISMISSED** [review]

**Review by guoen21 — COMMENTED** [review]
Need to fix the NPE crash issue.
Screen Shot 2024-01-30 at 11 08 31

**Review by guoen21 — APPROVED** [review]

**Review by WeiLiuSH — APPROVED** [review]

**Pull request merged**